src/main/java/de/tum/in/ase/parser/utils/XmlUtils.java

```diff
@@ -0,0 +1,65 @@
+package de.tum.in.ase.parser.utils;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+public class XmlUtils {
+    public static Optional<Element> getFirstChild(Element parent, String name) {
+        Iterator<Element> iterator = getChildElements(parent, name).iterator();
+        if (iterator.hasNext())
```

Comment:
please use braces { } here

Reply:
also in all other occurrences. It's bad practice to leave out braces in 1 line if / else statements